src/main/java/application/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,83 @@
+package application.controller;
+
+import application.dto.cart.item.CartItemRequestDto;
+import application.dto.shopping.cart.ShoppingCartRequestDto;
+import application.dto.shopping.cart.ShoppingCartResponseDto;
+import application.model.CartItem;
+import application.model.ShoppingCart;
+import application.model.User;
+import application.repository.CartItemRepository;
+import application.service.CartItemService;
+import application.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@Tag(name = "Management shopping carts", description = "Endpoints for managing shopping carts")
+@RequestMapping("/api/cart")
+public class ShoppingCartController {
+    private final CartItemRepository cartItemRepository;
+    private final CartItemService cartItemService;
+    private final ShoppingCartService shoppingCartService;
+
+    @PostMapping
+    @Operation(summary = "Add a book to the shopping cart",
+            description = "An endpoint for adding a book to the shopping cart")
+    @PreAuthorize("hasRole('USER')")

```

Comment:
Correct me if I'm wrong, but it looks like we use same authorization rules for all endpoints here. If so, we can move pre authorize annotation to the class level I guess